### PR TITLE
Gitlab support

### PR DESCRIPTION
**Pull request by napalm684**
Added support for Gitlab VCS/Gitlab CI

Happy to add docs once we confirm it is good.

Closes https://github.com/karlderkaefer/cdk-notifier/issues/40

### CODE REVIEW

**Comment by karlderkaefer** [comment]
looks good in general :+1: im checking the details this weekend

**Comment by napalm684** [comment]
Any questions issues or otherwise?

**Review by karlderkaefer — COMMENTED** [review]
code is good for merge, you can update Readme too.

File: cmd/root.go
Comment (karlderkaefer) [inline cmd/root.go:83]:
```suggestion
	rootCmd.Flags().String("vcs", "github", "Version Control System [github|bitbucket|gitlab]")
```

File: config/app.go
Comment (karlderkaefer) [inline config/app.go:67]: to reduce necessary cli args and fit better into structure, can you remove and load the `PROJ_ID` from API by using repo and owner? It would be also more natural for user the specify only owner and repo instead of the id
```
curl -XGET -H "Content-Type: application/json" --header "PRIVATE-TOKEN: $GITLAB_TOKEN" https://gitlab.com/api/v4/projects/pansen2%2Flearn-gitlab-ultimate-trial/
```

Comment (karlderkaefer) [reply config/app.go:67]: the Gitlab CI can provide these information too 
```
CI_PROJECT_NAME=learn-gitlab-ultimate-trial
CI_PROJECT_NAMESPACE=pansen2
```

**Comment by karlderkaefer** [comment]
I tested it locally, works for me :+1: I added two comments to review

**Comment by karlderkaefer** [comment]
@all-contributors please add @napalm684 for gitlab support

**Comment by karlderkaefer** [comment]
@all-contributors please add @karlderkaefer for creator

**Comment by karlderkaefer** [comment]
@all-contributors please add @napalm684 for code

File: config/app.go
Comment (napalm684) [reply config/app.go]: @karlderkaefer see what you think of this please.

**Comment by napalm684** [comment]
> code is good for merge, you can update Readme too.

Readme updated as well.

File: provider/gitlab.go
Comment (karlderkaefer) [inline provider/gitlab.go:80]: just for style I would like to return error earlier
```suggestion
	if err != nil {
		return "", err
	}

	return gc.ProjectId, nil
```

File: config/app.go
Comment (karlderkaefer) [reply config/app.go]: very good

**Comment by karlderkaefer** [comment]
@napalm684 look's great. I did one small comment, I still need to figure how to tell circleci to run the pipeline, would like to do that before merging. Ideally could you squash your commits following [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/)? Otherwise no new version is published. example 
````
feat(gitlab): add gitlab support 

closes #40 
```

**Comment by napalm684** [comment]
> @napalm684 look's great. I did one small comment, I still need to figure how to tell circleci to run the pipeline, would like to do that before merging. Ideally could you squash your commits following [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/)? Otherwise no new version is published. example
> 
> ```
> feat(gitlab): add gitlab support 
> 
> closes #40 
> ```

Squash and feedback completed. Let me know if there is anything you need help with on circleci. I haven't worked with it much but I am hoping to expedite getting this merged up. My devops group is waiting for the green light. Thanks!

**Review by karlderkaefer — APPROVED** [review]

**Comment by karlderkaefer** [comment]
found the issue the cirlceci pipleine, I'm using a secret in the steps for PRs. Im fixing this on main first

**Comment by karlderkaefer** [comment]
Im not sure why code scanner is pending at the moment, need to investigate

**Comment by karlderkaefer** [comment]
@napalm684 are you allowed to merge? if so you can merge now

**Comment by karlderkaefer** [comment]
sorry I messed up the merge, did rebase instead of merge, you are not listed as contributor, I need to revert the PR, I would kindly to ask to reopen the PR

**Pull request merged**
